Gems/Atom/RPI/Code/Tests/System/CullingTests.cpp

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Math/MatrixUtils.h>
+#include <AzCore/Task/TaskExecutor.h>
+#include <AzCore/Task/TaskGraph.h>
+#include <AzFramework/Visibility/OctreeSystemComponent.h>
+
+#include <Atom/RPI.Public/Culling.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Public/View.h>
+#include <Common/RPITestFixture.h>
+
+namespace UnitTest
+{
+    using namespace AZ;
+    using namespace RPI;
+
+    // The CullingTests fixture sets up a culling scene for testing culling.
+    // It also creates some views and a varying number of cullable objects visible in each view.
+    // It does not register the cullables with the culling scene, so their properties can be overridden
+    // before registering in order to test different scenarios
+    class CullingTests : public RPITestFixture
+    {
+        void SetUp() override
+        {
+            RPITestFixture::SetUp();
+
+            m_executor = aznew TaskExecutor{};
+            TaskExecutor::SetInstance(m_executor);
+
+            m_octreeSystemComponent = new AzFramework::OctreeSystemComponent;
```

Comment:
nit: Would be a little cleaner to put this in a unique ptr or just make it part of `CullingTests` itself